dialects/sqlite-3-38/src/main/kotlin/app/cash/sqldelight/dialects/sqlite_3_38/grammar/mixins/JsonExpressionMixin.kt

```diff
@@ -0,0 +1,20 @@
+package app.cash.sqldelight.dialects.sqlite_3_38.grammar.mixins
+
+import app.cash.sqldelight.dialects.sqlite_3_38.grammar.psi.SqliteJsonExpression
+import com.alecstrong.sql.psi.core.SqlAnnotationHolder
+import com.alecstrong.sql.psi.core.psi.SqlColumnDef
+import com.alecstrong.sql.psi.core.psi.SqlColumnName
+import com.alecstrong.sql.psi.core.psi.SqlCompositeElementImpl
+import com.intellij.lang.ASTNode
+
+internal abstract class JsonExpressionMixin(node: ASTNode) :
+  SqlCompositeElementImpl(node),
+  SqliteJsonExpression {
+  override fun annotate(annotationHolder: SqlAnnotationHolder) {
+    val columnType = ((firstChild.reference?.resolve() as? SqlColumnName)?.parent as? SqlColumnDef)?.columnType?.typeName?.text
+    if (columnType == null || columnType != "TEXT") {
```

Comment:
Should I add support for a `JSON` data type? I think it's supported as of 3.38 but the documentation isn't fully clear (from a quick look through).

Reply:
from my readthrough its just a shortcut for TEXT and I want to avoid messing with the SQLite types cause they're weird